### PR TITLE
feat: SSE

### PR DESCRIPTION
**Pull request by JamieSinn**

### CODE REVIEW

File: DevCycle.SDK.Server.Local/ConfigManager/EnvironmentConfigManager.cs [hunk 1]
```diff
@@ -131,9 +133,9 @@ private string GetConfigUrl()
          *
          * Unexpected exceptions will still be thrown from here.
          */
-        private async Task FetchConfigAsyncWithTask()
+        private async Task FetchConfigAsyncWithTask(uint lastmodified = 0)
```
Comment (jonathannorris) [inline DevCycle.SDK.Server.Local/ConfigManager/EnvironmentConfigManager.cs:138]: Is there any way to break this function into smaller parts? it's pretty long.

File: DevCycle.SDK.Server.Local/ConfigManager/EnvironmentConfigManager.cs [hunk 2]
```diff
@@ -149,68 +151,131 @@ private async Task FetchConfigAsyncWithTask()
...
+                        {
+                            logger.LogWarning(e, "Failed to parse SSE config. Skipping SSE Initialization");
+                        }
+
+                        localBucketing.StoreConfig(sdkKey, res.Content);
+                        configEtag = etag;
+                        configLastModified = lastModified;
+                        logger.LogDebug("Config successfully initialized with etag: {ConfigEtag}, {lastmodified}",
+                            configEtag, configLastModified);
+                        //eventQueue?.QueueSDKConfigEvent(request, res);
```
Comment (jonathannorris) [inline DevCycle.SDK.Server.Local/ConfigManager/EnvironmentConfigManager.cs]: cleanup?

Comment (JamieSinn) [reply DevCycle.SDK.Server.Local/ConfigManager/EnvironmentConfigManager.cs]: yeah - was gonna stack sdkconfig on top of this

File: DevCycle.SDK.Server.Local/ConfigManager/EnvironmentConfigManager.cs [hunk 3]
```diff
@@ -149,68 +151,131 @@ private async Task FetchConfigAsyncWithTask()
...
+        private void SSEStateHandler(object sender, StateChangedEventArgs args)
+        {
+            switch (args.ReadyState)
+            {
+                case ReadyState.Raw:
+                    break;
+                case ReadyState.Connecting:
+                    break;
+                case ReadyState.Open:
+                    var ssepollingIntervalMs = 60 * 60 * 15 * 1000;
```
Comment (jonathannorris) [inline DevCycle.SDK.Server.Local/ConfigManager/EnvironmentConfigManager.cs]: this should probably be a static const at the top?

File: DevCycle.SDK.Server.Local/ConfigManager/EnvironmentConfigManager.cs (hunk 1 above)
Comment (JamieSinn) [reply DevCycle.SDK.Server.Local/ConfigManager/EnvironmentConfigManager.cs:138]: It really just looks long - it's 110 lines with formatting and braces - it's about 80 lines of code.

File: DevCycle.SDK.Server.Local/ConfigManager/EnvironmentConfigManager.cs (hunk 3 above)
Comment (JamieSinn) [reply DevCycle.SDK.Server.Local/ConfigManager/EnvironmentConfigManager.cs]: yeah - just currently messing around with some timings so i had it there - will move

File: DevCycle.SDK.Server.Local/ConfigManager/EnvironmentConfigManager.cs [hunk 4]
```diff
@@ -33,22 +35,24 @@ public class EnvironmentConfigManager : IDisposable
...
-
+        private bool pollingEnabled = true;
+        private SSEManager sseManager;
         private string configEtag = "";
         private string configLastModified = "";
 
+        public virtual string Config { get; private set; }
+        public virtual bool Initialized { get; internal set; }
+        
+        private const int ssepollingIntervalMs = 15 * 60 * 60 * 1000;
```
Comment (jonathannorris) [inline DevCycle.SDK.Server.Local/ConfigManager/EnvironmentConfigManager.cs]: `ssePollingIntervalMs`

**Review by jonathannorris — APPROVED** [review]

**Pull request merged**